tsl/src/deparse.c

```diff
@@ -1003,3 +1003,88 @@ deparse_grant_revoke_on_database(Node *node, const char *dbname)
 
 	return command->data;
 }
+
+/* Deparse user-defined trigger */
+const char *
+deparse_create_trigger(CreateTrigStmt *stmt)
+{
+	ListCell *lc;
+	bool found_event = false;
+	bool found_first_arg = false;
+
+	/*
+	CREATE [ CONSTRAINT ] TRIGGER name { BEFORE | AFTER | INSTEAD OF } { event [ OR ... ] }
+		ON table_name
+		[ FROM referenced_table_name ]
+		[ NOT DEFERRABLE | [ DEFERRABLE ] [ INITIALLY IMMEDIATE | INITIALLY DEFERRED ] ]
+		[ REFERENCING { { OLD | NEW } TABLE [ AS ] transition_relation_name } [ ... ] ]
+		[ FOR [ EACH ] { ROW | STATEMENT } ]
+		[ WHEN ( condition ) ]
+		EXECUTE { FUNCTION | PROCEDURE } function_name ( arguments )
+	*/
+	if (stmt->isconstraint)
+		elog(ERROR, "deparsing constraint triggers is not supported");
+
+	StringInfo command = makeStringInfo();
+	appendStringInfo(command, "CREATE TRIGGER %s ", quote_identifier(stmt->trigname));
+
+	if (TRIGGER_FOR_BEFORE(stmt->timing))
+		appendStringInfoString(command, "BEFORE");
+	else if (TRIGGER_FOR_AFTER(stmt->timing))
+		appendStringInfoString(command, "AFTER");
+	else if (TRIGGER_FOR_INSTEAD(stmt->timing))
+		appendStringInfoString(command, "INSTEAD OF");
+	else
+		elog(ERROR, "unexpected timing value: %d", stmt->timing);
```

Comment:
Just some nits, since this seems to be the style you adopted for building the string:
```suggestion
	appendStringInfo(command, "CREATE TRIGGER %s", quote_identifier(stmt->trigname));

	if (TRIGGER_FOR_BEFORE(stmt->timing))
		appendStringInfoString(command, " BEFORE");
	else if (TRIGGER_FOR_AFTER(stmt->timing))
		appendStringInfoString(command, " AFTER");
	else if (TRIGGER_FOR_INSTEAD(stmt->timing))
		appendStringInfoString(command, " INSTEAD OF");
	else
		elog(ERROR, "unexpected timing value: %d", stmt->timing);
```

Reply:
The reasoning here was that I wanted to avoid typing space characters, to minimize duplication. Whenever I could have a space character in one place instead of several I would change style.

Reply:
This is just a nit. I leave it up to you to decide if you want to use my suggestion or not.